File: app/src/main/java/com/ch/yoon/kakao/pay/imagesearch/data/repository/ImageRepository.java
package com.ch.yoon.kakao.pay.imagesearch.data.repository;

import androidx.annotation.NonNull;

import com.ch.yoon.kakao.pay.imagesearch.data.local.room.entity.SearchLog;
import com.ch.yoon.kakao.pay.imagesearch.data.model.imagesearch.request.ImageSearchRequest;
import com.ch.yoon.kakao.pay.imagesearch.data.model.imagesearch.response.ImageSearchResponse;
import com.ch.yoon.kakao.pay.imagesearch.data.model.imagesearch.response.ImageSearchResult;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

/**
 * Creator : ch-yoon
 * Date : 2019-08-01.
 */
public interface ImageRepository {

    @NonNull
    Single<ImageSearchResult> requestImageList(@NonNull ImageSearchRequest imageSearchRequest);

    @NonNull
    Completable deleteSearchLog(@NonNull String keyword);

    @NonNull
    Single<List<SearchLog>> requestSearchLogList();

    @NonNull
    Single<SearchLog> insertOrUpdateSearchLog(@NonNull String keyword);

}

File: app/src/main/java/com/ch/yoon/kakao/pay/imagesearch/data/repository/image/ImageRepository.kt
package com.ch.yoon.kakao.pay.imagesearch.data.repository.image

import com.ch.yoon.kakao.pay.imagesearch.data.source.remote.kakao.request.ImageSearchRequest
import com.ch.yoon.kakao.pay.imagesearch.data.repository.image.model.ImageSearchResponse
import com.ch.yoon.kakao.pay.imagesearch.data.repository.image.model.SearchLog
import io.reactivex.Completable
import io.reactivex.Single

/**
 * Creator : ch-yoon
 * Date : 2019-10-28
 **/
interface ImageRepository {

    fun requestImageList(imageSearchRequest: ImageSearchRequest): Single<ImageSearchResponse>

    fun deleteSearchLog(searchLog: SearchLog): Completable

    fun requestSearchLogList(): Single<List<SearchLog>>

    fun insertOrUpdateSearchLog(keyword: String): Single<SearchLog>

}